spec.html

```diff
@@ -37,8 +37,75 @@
   </li>
 </ul>
 
-<!-- TODO -->
+<hr>
+
+<p><ins class="block"><a href="https://tc39.github.io/ecma262/#sec-characterclassescape">21.2.2.12 CharacterClassEscape</a> is changed as follows:</ins></p>
+
+<p>The production <emu-grammar>CharacterClassEscape :: `p{` UnicodePropertyValueExpression `}`</emu-grammar> evaluates <del>by returning the CharSet containing all Unicode code points included in the CharSet returned by |UnicodePropertyValueExpression|</del> <ins>as follows:</ins></p>
+<p><ins>The production <emu-grammar>CharacterClassEscape :: `p{` UnicodePropertyValueExpression `}`</emu-grammar> evaluates as follows:</ins></p>
+<emu-alg>
+  1. <ins>Let _v_ be the return value of |UnicodePropertyValueExpression|.</ins>
+  1. <ins>If _v_ is a CharSet, then</ins>
+    1. <ins>Return the CharSet containing all Unicode code points included in _v_.</ins>
+  1. <ins>Assert: _v_ is a SequenceSet.</ins>
+  1. <ins>Return the Disjunction containing an Alternative for each of the Unicode code point sequences in _v_.</ins><!-- TODO: Improve this. -->
+</emu-alg>
+
+<p>The production <emu-grammar>CharacterClassEscape :: `P{` UnicodePropertyValueExpression `}`</emu-grammar> evaluates by returning the CharSet containing all Unicode code points not included in the CharSet returned by |UnicodePropertyValueExpression|.</p>
+
+<p>The production <emu-grammar>UnicodePropertyValueExpression :: LoneUnicodePropertyNameOrValue</emu-grammar> evaluates as follows:</p>
+<emu-alg>
+  1. Let _s_ be SourceText of |LoneUnicodePropertyNameOrValue|.
+  1. If ! UnicodeMatchPropertyValue(`"General_Category"`, _s_) is identical to a List of Unicode code points that is the name of a Unicode general category or general category alias listed in the &ldquo;Property value and aliases&rdquo; column of <a href="https://tc39.github.io/ecma262/#table-unicode-general-category-values">Table 53</a>, then
+    1. Return the CharSet containing all Unicode code points whose character database definition includes the property &ldquo;General_Category&rdquo; with value _s_.
+  1. <ins>If _s_ is identical to a List of Unicode code points that is the name of a Unicode sequence property or sequence property alias listed in the &ldquo;Property value and aliases&rdquo; column of <emu-xref href="#table-unicode-sequence-properties"></emu-xref>, then</ins>
+    1. <ins>Return the SequenceSet containing each Unicode code point sequence included in the Unicode property _s_.</ins>
+  1. Let _p_ be ! UnicodeMatchProperty(_s_).
+  1. Assert: _p_ is a binary Unicode property or binary property alias listed in the &ldquo;Property name and aliases&rdquo; column of <a href="https://tc39.github.io/ecma262/#table-binary-unicode-properties">Table 52</a>.
+  1. Return the CharSet containing all Unicode code points whose character database definition includes the property _p_ with value &ldquo;True&rdquo;.
```

Comment:
It’s a little annoying that it now returns either a `SequenceSet` or a `CharSet`.